### PR TITLE
homework number thirteen

### PR DESCRIPTION
**Pull request by kihors**

### CODE REVIEW

File: src/com/homework/thirteenth/Animal.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public boolean equals(Object obj) {
+        if (obj instanceof Animal) {
+            return Objects.equals(name, ((Animal) obj).name);
+        }
+        return false;
+    }
+
+    public static Animal[] deserializeAnimal(byte[] data) {
+        int countOfAnimals;
+        Animal[] animals;
```
Comment (Sofasmile) [inline src/com/homework/thirteenth/Animal.java]: better move this variables in block try

File: src/com/homework/thirteenth/Animal.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            countOfAnimals = objectInputStream.readInt();
+            animals = new Animal[countOfAnimals];
+            for (int i = 0; i < countOfAnimals; i++) {
+                Animal animal = (Animal) objectInputStream.readObject();
+                animals[i] = animal;
+            }
+        } catch (Exception e) {
+            throw new IllegalArgumentException();
+        }
+        return animals;
```
Comment (Sofasmile) [inline src/com/homework/thirteenth/Animal.java]: move return in try block

File: src/com/homework/thirteenth/Animal.java (hunk 1 above)
Comment (kihors) [reply src/com/homework/thirteenth/Animal.java]: done

File: src/com/homework/thirteenth/Animal.java (hunk 2 above)
Comment (kihors) [reply src/com/homework/thirteenth/Animal.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**